json-schema/json-schema-generic/src/main/scala/endpoints/generic/JsonSchemas.scala

```diff
@@ -51,59 +50,56 @@ trait JsonSchemas extends algebra.JsonSchemas {
+      (docs: HNil) => emptyRecord.xmap[HNil](_ => HNil)(_ => ())
 
     implicit def singletonCoproduct[L <: Symbol, A](implicit
       labelSingleton: Witness.Aux[L],
       recordA: GenericRecord[A]
     ): GenericTagged[FieldType[L, A] :+: CNil] =
-      new GenericTagged[FieldType[L, A] :+: CNil] {
-        def jsonSchema: Tagged[FieldType[L, A] :+: CNil] =
+      new GenericTagged[FieldType[L, A] :+: CNil](
+        jsonSchema =
           recordA.jsonSchema.tagged(labelSingleton.value.name).xmap[FieldType[L, A] :+: CNil] {
             a => Inl(shapelessField[L](a))
           } {
             case Inl(a) => a
             case Inr(_) => sys.error("Unreachable code")
           }
-      }
+    )
 
   }
 
   trait GenericJsonSchemaLowPriority extends GenericJsonSchemaLowLowPriority {
 
-    implicit def consRecord[L <: Symbol, H, T <: HList](implicit
+    implicit def consRecord[L <: Symbol, H, T <: HList, DH <: Option[docs], DT <: HList](implicit
       labelHead: Witness.Aux[L],
       jsonSchemaHead: JsonSchema[H],
-      jsonSchemaTail: GenericRecord[T]
-    ): GenericRecord[FieldType[L, H] :: T] =
-      new GenericRecord[FieldType[L, H] :: T] {
-        def jsonSchema: Record[FieldType[L, H] :: T] =
-          (field(labelHead.value.name)(jsonSchemaHead) zip jsonSchemaTail.jsonSchema)
+      jsonSchemaTail: DocumentedGenericRecord[T, DT]
+    ): DocumentedGenericRecord[FieldType[L, H] :: T, DH :: DT] =
+      new DocumentedGenericRecord[FieldType[L, H] :: T, DH :: DT] {
+        def record(docs: DH :: DT) =
+          (field(labelHead.value.name, docs.head.map(_.text))(jsonSchemaHead) zip jsonSchemaTail.record(docs.tail))
             .xmap[FieldType[L, H] :: T] { case (h, t) => shapelessField[L](h) :: t }(ht => (ht.head, ht.tail))
       }
 
-    implicit def consOptRecord[L <: Symbol, H, T <: HList](implicit
+    implicit def consOptRecord[L <: Symbol, H, T <: HList, DH <: Option[docs], DT <: HList](implicit
       labelHead: Witness.Aux[L],
       jsonSchemaHead: JsonSchema[H],
-      jsonSchemaTail: GenericRecord[T]
-    ): GenericRecord[FieldType[L, Option[H]] :: T] =
-      new GenericRecord[FieldType[L, Option[H]] :: T] {
-        def jsonSchema: Record[FieldType[L, Option[H]] :: T] =
-          (optField(labelHead.value.name)(jsonSchemaHead) zip jsonSchemaTail.jsonSchema)
+      jsonSchemaTail: DocumentedGenericRecord[T, DT]
+    ): DocumentedGenericRecord[FieldType[L, Option[H]] :: T, DH :: DT] =
+      new DocumentedGenericRecord[FieldType[L, Option[H]] :: T, DH :: DT] {
+        def record(docs: DH :: DT) =
+          (optField(labelHead.value.name, docs.head.map(_.text))(jsonSchemaHead) zip jsonSchemaTail.record(docs.tail))
             .xmap[FieldType[L, Option[H]] :: T] { case (h, t) => shapelessField[L](h) :: t }(ht => (ht.head, ht.tail))
       }
 
-    implicit def consCoproduct[L <: Symbol, H, T <: Coproduct](implicit
+    implicit def consCoproduct[L <: Symbol, H, T <: Coproduct, D](implicit
```

Comment:
what's the new type param `D` for? I can't find its actual usage.

Reply:
Good catch, that’s for the next PR ;) I’ll remove it.